engine/execution/state/state.go

```diff
@@ -408,73 +408,71 @@ func (s *state) saveExecutionResults(
 		return fmt.Errorf("can not retrieve chunk data packs: %w", err)
 	}
 
-	err = s.chunkDataPacks.Store(chunks)
+	err = storage.WithLock(s.lockManager, storage.LockInsertChunkDataPack, func(lctx lockctx.Context) error {
```

Comment:
This process acquires two locks, it's better to use `WithLock` function instead of AcquireLock, because a misuse of `defer lctx.Release` will hold the lock for too long.